cluster/core/src/main/scala/com/lightbend/lagom/internal/cluster/JoinClusterImpl.scala

```diff
@@ -42,12 +42,33 @@ private[lagom] object JoinClusterImpl {
       )
     }
 
-    if (clusterBootstrapEnabled) {
-      // TODO: move AkkaManagement to Guice module
-      AkkaManagement(system.asInstanceOf[ExtendedActorSystem]).start()
-      ClusterBootstrap(system.asInstanceOf[ExtendedActorSystem]).start()
-    } else if (cluster.settings.SeedNodes.isEmpty && joinSelf) {
-      cluster.join(cluster.selfAddress)
+    /*
+     * There are four ways to form a cluster in Lagom
+     *  1. lagom.cluster.join-self - Lagom's dev mode forms a single node cluster
+     *  2. lagom.cluster.bootstrap.enabled - Lagom's prod default. Uses Akka Cluster Bootstrap and Akka Management
+     *  3. Declared seed-nodes. Overrides the optoins above
+     *  4. Programmatically: No seed-nodes, join-self = false and bootstrap.enabled = false. User is on its own to form the cluster
+     *
+     *  Last option is rather unusual, but we should not block users willing to do so. We also use that option in multi-jvm tests.
+     *  In order to join programmatically, one need to disable all flags.
+     *
```

Comment:
I find the ordering in the comment to be inconsistent with the actual order of precedence. How about:

```
    /*
     * There are several ways to form a cluster in Lagom
     *  1. Declared seed-nodes. Highest priority setting.
     *  2. lagom.cluster.bootstrap.enabled - Lagom's prod default. Uses Akka Cluster Bootstrap and Akka Management
     *  3. lagom.cluster.join-self - Lagom's dev mode forms a single node cluster (unrecommended for production)
     *  4. Programmatically: No seed-nodes, join-self = false and bootstrap.enabled = false. User is on its own to form the cluster
     *
     *  Last option is rather unusual but allows users to fallback to plain Akka cluster formation API. This option
     *  is used in Lagom tests, for example.
     *  In order to join programmatically, one need to disable all flags.
     *
     *  The code below will form the cluster if there are no seed-nodes and bootstrap or join-self are enabled. Forming
     *  the cluster when seed-nodes are defined is already handled by Akka internally so there's no need to add 
     *  extra code in Lagom's codebase. 
     */
```

So the abode suggestion reorders items `1.`, `2.` and `3.` to reflect precedence and rewrites the extra comments.

Reply:
sure